docs/architecture.md

```diff
@@ -1,188 +1,264 @@
 
-`policy-server` performs the validation of policies's settings by
-invoking the `validate_setting` function exposed by each policy.
-This topic is covered more in depth inside
-of [this section](/writing-policies/spec/01-intro-spec.md) of the documentation.
+The `policy-server` exits with an error if one or more policies
+received wrong configuration parameters from the end-user.
 
-`policy-server` will exit with an error if one or more policies received wrong
-configuration parameters from the end user.
+<!--TODO:Can we show the pool of worker threads in the diagram? Or maybe
+it's the stacked things?-->
 
-If all the policies are properly configured, `policy-server` will spawn a
-pool of worker threads to evaluate incoming requests using the Kubewarden
-policies specified by the user.
+If all policies are correctly configured,
+`policy-server`
+spawns a pool of worker threads
+to evaluate incoming requests
+using the Kubewarden policies
+specified by the user.
 
-Finally, `policy-server` will start a HTTPS server that listens to incoming
-validation requests. The web server is secured using the TLS key and certificate
-that have been previously created by `kubewarden-controller`.
+<!--TODO:Is the HTTPS server referred to, that which is in the aquamarine
+box, the 'Kubewarden Policy Server-->
 
-Each policy is exposed by the web server via a dedicated path that follows this
-naming convention: `/validate/<policy ID>`.
+Finally, the `policy-server` starts a HTTPS server
+listening to incoming validation requests.
+Kubewarden uses the TLS key and certificate
+created by `kubewarden-controller`
+to secure the web server.
 
-This is how the cluster looks like once the initialization of `policy-server`
-is completed:
+The web server exposes each policy by a dedicated path
+following the naming convention: `/validate/<policy ID>`.
+
+This diagram shows the cluster when initialization of `policy-server` is complete:
 
 ![policy-server initialized](/img/architecture_sequence_03.png)
 
 ### Making Kubernetes aware of the policy
 
+<!--TODO:Next paragraph. Says 'admission reviews', should that maybe be
+'admission requests'-->
+
 The `policy-server` Pods have a
 [`Readiness Probe`](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/),
-`kubewarden-controller` relies on that to know when the `policy-server` Deployment
-is ready to evaluate admission reviews.
+that `kubewarden-controller` uses to check when
+the `policy-server` Deployment is ready to evaluate admission reviews.
+
+<!--TODO:Aware? Is notified? Notifies? Is 'notifies' technically correct? Messages?-->
 
-Once the `policy-server` Deployment is marked as `Ready`, `kubewarden-controller`
-will make the Kubernetes API server aware of the new policy by creating either a
-`MutatingWebhookConfiguration` or a `ValidatingWebhookConfiguration`
-object.
+Once the readiness probe marks a `policy-server` Deployment as `Ready`,
```

Comment:
I would prefer to leave this phrase as it was, as the readiness probe doesn't necessarily mark the the Deployment as Ready, is the k8s reconciler that does.

Reply:
oj, reverted.